pallets/pallet-bonded-coins/src/benchmarking.rs

```diff
@@ -1 +1,855 @@
+		let pool_account = pool_id.clone().into();
+
+		T::CollateralCurrencies::touch(collateral_id.clone(), &pool_account, &account_origin)
+			.expect("Touching should work");
+
+		set_collateral_balance::<T>(collateral_id, &pool_account, 10000u128);
+
+		let beneficiary = AccountIdLookupOf::<T>::from(account_origin.clone());
+		let amount_to_burn = 10u128;
+
+		#[extrinsic_call]
+		burn_into(
+			origin as T::RuntimeOrigin,
+			pool_id,
+			0,
+			beneficiary,
+			amount_to_burn.saturated_into(),
+			0u128.saturated_into(),
+			T::MaxCurrencies::get(),
+		);
+
+		let balance = T::Fungibles::balance(target_asset_id, &account_origin);
+		assert_eq!(balance, (start_balance - amount_to_burn).saturated_into());
+	}
+
+	#[benchmark]
+
+	fn burn_into_lmsr(c: Linear<1, { <T as Config>::MaxCurrencies::get() }>) {
+		let origin = T::PoolCreateOrigin::try_successful_origin().expect("creating origin should not fail");
+		let account_origin = origin
+			.clone()
+			.into_signer()
+			.expect("generating account_id from origin should not fail");
+		make_free_for_deposit::<T>(&account_origin);
+
+		let collateral_id = create_default_collateral_asset::<T>();
+		let bonded_currencies = create_bonded_currencies_in_range::<T>(c, false);
+		let target_asset_id = bonded_currencies[0].clone();
+
+		let start_balance = 100u128;
+		set_fungible_balance::<T>(target_asset_id.clone(), &account_origin, start_balance);
+
+		let curve = get_lmsr_curve::<CurveParameterTypeOf<T>>();
+		let pool_id = create_pool::<T>(curve, bonded_currencies, None, None, Some(0));
+		let pool_account = pool_id.clone().into();
+
+		T::CollateralCurrencies::touch(collateral_id.clone(), &pool_account, &account_origin)
+			.expect("Touching should work");
+
+		set_collateral_balance::<T>(collateral_id, &pool_account, 10000u128);
+
+		let beneficiary = AccountIdLookupOf::<T>::from(account_origin.clone());
+		let amount_to_burn = 10u128;
+
+		#[extrinsic_call]
+		burn_into(
+			origin as T::RuntimeOrigin,
+			pool_id,
+			0,
+			beneficiary,
+			amount_to_burn.saturated_into(),
+			0u128.saturated_into(),
+			T::MaxCurrencies::get(),
+		);
+
+		let balance = T::Fungibles::balance(target_asset_id, &account_origin);
+		assert_eq!(balance, (start_balance - amount_to_burn).saturated_into());
+	}
+
+	#[benchmark]
+	fn start_destroy(c: Linear<1, { <T as Config>::MaxCurrencies::get() }>) {
+		let origin = T::DefaultOrigin::try_successful_origin().expect("creating origin should not fail");
+		let account_origin = origin
+			.clone()
+			.into_signer()
+			.expect("generating account_id from origin should not fail");
+
+		let bonded_currencies = create_bonded_currencies_in_range::<T>(c, false);
+		let curve = get_lmsr_curve::<CurveParameterTypeOf<T>>();
+
+		let pool_id = create_pool::<T>(
+			curve,
+			bonded_currencies.clone(),
+			Some(account_origin.clone()),
+			None,
+			None,
+		);
+
+		#[extrinsic_call]
+		_(origin as T::RuntimeOrigin, pool_id.clone(), T::MaxCurrencies::get());
```

Comment:
No CloNe EithEr!11!

Reply:
no clone, function calls, or conversations in tx: [44841bd](https://github.com/KILTprotocol/kilt-node/pull/789/commits/44841bdaa050c2179d0748ac59fbf9357845197f)